File: py/two-sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash = dict()
        for i in range(len(nums)):
            v = nums[i]
            if hash.get(target-v) != None: return [hash.get(target-v), i]
            hash[v] = i
        return None

File: go/two-sum.go
func twoSum(nums []int, target int) []int {
    hash := make(map[int]int)
    for i,v := range(nums) {
        if _, ok := hash[target-v]; ok {
            return []int { hash[target-v], i }
        }
        hash[v] = i
    }
    return nil
}
